lib/ipmi_fru.c

```diff
@@ -596,15 +608,14 @@ write_fru_area(struct ipmi_intf * intf, struct fru_info *fru, uint8_t id,
 			break;
 		}
 
-		if (rsp->ccode == 0xc7 || rsp->ccode == 0xc8 || rsp->ccode == 0xca) {
-			if (fru->max_write_size > 8) {
-				fru->max_write_size -= 8;
-				lprintf(LOG_INFO, "Retrying FRU write with request size %d",
-						fru->max_write_size);
-				continue;
-			}
-		} else if(rsp->ccode == 0x80) {
-			rsp->ccode = 0;
+		if (fru_cc_rq2big(rsp->ccode) && fru->max_write_size > FRU_AREA_MAXIMUM_BLOCK_SZ) {
+			fru->max_write_size -= FRU_BLOCK_SZ;
```

Comment:
I merged these two conditions, which is a change unrelated to any commit message.